Compiling with --library fails when using the LLVM back end

### Summary of Problem
The following two tests fail when testing with \-compopts \-\-llvm.
```
compflags/library/withMain/lib2
compflags/lydia/library/errorMessage/hasMain
```
Both of these tests are trying to compile with \-\-library.  This results in a failure like the following when CHPL_TARGET_COMPILER=gnu.
```
[Executing diff lib2.good lib2.1.comp.out.tmp]
1a2,5
> gcc: warning: /tmp/chpl-c17571-23924.deleteme/chpl__module.bc: linker input file unused because linking not done
> ar: /tmp/chpl-c17571-23924.deleteme/liblib.tmp.a: File format not recognized
> gmake: *** [/tmp/chpl-c17571-23924.deleteme/liblib.tmp.a] Error 1
> error: Make Binary - Building Launcher and Copying
[Error matching compiler output for compflags/library/withMain/lib2]
```
The failure mode is different when CHPL_TARGET_COMPILER=clang.
```
[Executing diff lib2.good lib2.1.comp.out.tmp]
1a2,21
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/third-party/qthread/install/darwin-clang-native-flat/include' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/third-party/hwloc/install/darwin-clang-native-flat/include' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/third-party/qthread/install/darwin-clang-native-flat/include' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I .' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/localeModels/flat' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/localeModels' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/comm/none' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/comm' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/tasks/qthreads' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/threads/none' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/qio' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/atomics/cstdlib' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime/include/mem/jemalloc' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/third-party/utf8-decoder' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/runtime//../build/runtime/darwin/clang/arch-native/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/jemalloc/atomics-cstdlib/gmp-none/hwloc/re2/wide-struct/fs-none/include' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/third-party/jemalloc/install/darwin-clang-native/include' [-Werror,-Wunused-command-line-argument]
> clang: error: argument unused during compilation: '-I /Users/dmk/chapel/third-party/hwloc/install/darwin-clang-native-flat/include' [-Werror,-Wunused-command-line-argument]
> make: *** [/var/folders/9t/79lmc8gs4lg_jw7lx22m52hr0006rl/T//chpl-dmk-33468.deleteme/liblib.tmp.a] Error 1
> error: Make Binary - Building Launcher and Copying
[Error matching compiler output for compflags/library/withMain/lib2]
```
### Steps to Reproduce

**Source Code:**
See the tests `compflags/library/withMain/lib2` and
`compflags/lydia/library/errorMessage/hasMain`.

**Compile command:**
```sh
$ cd $CHPL_HOME/test
$ ../util/start_test -compopts --llvm compflags/library/withMain/lib2.chpl
$ ../util/start_test -compopts --llvm compflags/lydia/library/errorMessage/hasMain.chpl
```

**Execution command:**
Same as above

**Associated Future Test(s):**
No futures, just these regular tests.
```
compflags/library/withMain/lib2
compflags/lydia/library/errorMessage/hasMain
```
### Configuration Information

- Output of `chpl --version`:
```
chpl Version 1.16.0 pre-release (8aceb8c99d)
Copyright (c) 2004-2017, Cray Inc.  (See LICENSE file for more details)
```
- Output of `$CHPL_HOME/util/printchplenv --anonymize --debug`:

For gcc:
```
CHPL_HOST_PLATFORM: linux64 *
CHPL_HOST_COMPILER: gnu
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
  CHPL_COMM_SUBSTRATE: none
  CHPL_GASNET_SEGMENT: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
  CHPL_JEMALLOC: jemalloc
CHPL_MAKE: gmake
CHPL_ATOMICS: intrinsics
CHPL_GMP: none *
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_WIDE_POINTERS: struct
CHPL_LLVM: llvm *
CHPL_AUX_FILESYS: none
```
For clang:
```
CHPL_HOST_PLATFORM: darwin *
CHPL_HOST_COMPILER: clang
CHPL_TARGET_PLATFORM: darwin
CHPL_TARGET_COMPILER: clang
CHPL_TARGET_ARCH: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
  CHPL_COMM_SUBSTRATE: none
  CHPL_GASNET_SEGMENT: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
  CHPL_JEMALLOC: jemalloc
CHPL_MAKE: make
CHPL_ATOMICS: cstdlib *
CHPL_GMP: none *
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_WIDE_POINTERS: struct
CHPL_LLVM: llvm *
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:

For gcc:
```
gcc (GCC) 7.1.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
For clang:
```
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
- (For Cray systems only) Output of `module list`:
N/A

--- a/test/compflags/library/withMain/lib2.suppressif
+++ b/test/compflags/library/withMain/lib2.suppressif
@@ -1,7 +1,0 @@
-# Known LLVM issue with --library
-#
-# This is a known issue where we are not yet able to handle compiling
-# with --library when using the LLVM back end.  See github issue
-# #6336.  Suppress this for now.
-
-CHPL_LLVM!=none

--- a/test/compflags/lydia/library/errorMessage/hasMain.suppressif
+++ b/test/compflags/lydia/library/errorMessage/hasMain.suppressif
@@ -1,7 +1,0 @@
-# Known LLVM issue with --library
-#
-# This is a known issue where we are not yet able to handle compiling
-# with --library when using the LLVM back end.  See github issue
-# #6336.  Suppress this for now.
-
-CHPL_LLVM!=none

